### ISSUE TITLE
Review app for low data, lower-spec device users

### ISSUE THREAD

**Issue by fancyham** [hackforla/food-oasis#928]
Does app work with challenging data situations and with our target user's minimum specs?

We need to check on this and confirm our web app is fast, stable, and uses as little data as possible.

We should do this periodically and when we roll out major updates that our app works for our users who may have limited data.

**Discussion:**

Our target users may be on spotty wifi or have cellular data plan limits.  Plus reducing page load sizes makes the app more responsive.

By optimizing for lower-end devices and data plans, we make the app better for everyone.

**How to test:**
- Measure data transferred in developer console on a web browser.

Nice to have:
- Some web browsers have a 'responsive' mode for developers to simulate a smaller screen.  Some also have bandwidth controls to simulate slow networks.  That in combination with a bandwidth limiter app may help simulate.

Optional:
- Test on phones our users may be using and measure data use and responsiveness.  Phones may be a bit older and have smaller screens.
- Try it somewhere with bad wifi or set your phone to 3G.
- Partner with an organization that works with our target audience and ask them to test it.

For throttling, see: https://blog.bam.tech/developer-news/simulate-poor-network-test-mobile-apps-device
and https://www.google.com/search?q=throttle+bandwidth+simulate+phone+performance

- [x] Audit site with target users, those devices, or via simulation
- [ ] (if possible) Try to identify target user's device and bandwidth specs (desk research of literature, surveys?)
- [ ] (if possible) Try to identify current user population's device and bandwidth specs (via analytics?)

**Comment by fancyham** [comment]
If anyone is curious about California lifeline phones, check out this link. The comments are great - gruff but helpful. Also gives you a sense of the issues people have with their phones.

TL;DR: One per household no matter how many in household, most have limited data with many in 3-5GB range, all are smartphones now.

https://www.freegovernmentcellphones.net/states/california-government-cell-phone-providers

thanks for the link, @wanderingspoon

**Comment by fancyham** [comment]
Use the responsive simulation mode (in Safari: Developer > Enter Responsive Design Mode) to simulate different devices and use the web inspector > Network to get bandwidth used and time elapsed.

Create a spreadsheet:

Title: Put the date, your name, what browser and computer you're using, your location

Columns: Landing page, first map, organization detail, About Us, FAQ, etc.
Rows: Device and screens size (use developer responsive mode)
Cells: Amount of data needed to load the entire page, and optionally, the time it took to load (use the developer console)

**Comment by staceyrebekahscott** [comment]
Check in with Slack channels: UI-UX, Data-Science, Developers
Ask about HfLA guides or frameworks for testing 
And check the Community of Practice kanbans for issues.

**Comment by fancyham** [comment]
Just realized that this issue is really only about data transferred — instructions for how to do it are above, but for Accessibility, please see issue #1355

**Comment by fancyham** [comment]
Please try to add screenshots of the list of files being transferred so that we can identify which items are using the most data and to see if we're still needing those items. (i.e. hotjar.com)
![Screen Shot 2022-10-01 at 2 16 22 PM](https://user-images.githubusercontent.com/3376957/193428604-9a298702-610a-4af9-b779-436c36e2abfd.jpg)

**Comment by jnaito777** [comment]
List of file names transferred. The first two screenshots show the full list of files being transferred. The last screen shot is sorted by transfer size.

<img width="816" alt="Screen Shot 2022-10-03 at 4 02 27 PM" src="https://user-images.githubusercontent.com/114040198/193700954-7a68f231-5e2f-4e24-b801-c631d8db1b78.png">
<img width="810" alt="Screen Shot 2022-10-03 at 4 02 41 PM" src="https://user-images.githubusercontent.com/114040198/193700962-064b6e07-0110-416c-ab26-41f594c4c490.png">
<img width="809" alt="Screen Shot 2022-10-03 at 4 51 41 PM" src="https://user-images.githubusercontent.com/114040198/193705796-db8a4f6f-9b0f-4b19-8218-8d316408fc84.png">

**Comment by jnaito777** [comment]
Attached the results and created a guide to be able to recreate this simulation in the future. This accessibility bench mark covered various devices (mobile, tablet, laptop) with various sizes via responsive mode, and measured resource size, transfer size, and load time. Also performed a simulation benchmark on how long (theoretically) it takes for a user to navigate the website in various network connections.

Through my research, users who have poor network connections have significantly longer load times.

[UX Accessibility Benchmarking on Safari Guide.pdf](https://github.com/hackforla/food-oasis/files/9728435/UX.Accessibility.Benchmarking.on.Safari.Guide.pdf)
[UX Accessibility Benchmarking.pdf](https://github.com/hackforla/food-oasis/files/9728438/UX.Accessibility.Benchmarking.pdf)

**Comment by fancyham** [comment]
One thing I noticed here are that we have a hotjar.com script (it records clicks and movements) but I don’t think we have access to that account any more. That would be something to remove.

- [ ] Investigate removing hotjar.com scripts if we are not using its analytics

**Comment by fancyham** [comment]
- [ ] Reduce the size of images on static pages (about us, donate, FAQ) so they load faster and use less data
- [ ] If possible, create 'loading' progress indicator during initial page load (while React and other javascript components are being loaded)
- [ ] Check for unused scripts (such as hotjar.com) and remove them
